pkg/util/http/transport.go

```diff
@@ -125,17 +140,36 @@ func GetProxyTransportFunc(p *config.Proxy) func(*http.Request) (*url.URL, error
 			return nil, nil
 		}(r)
 
-		// Print a warning if the proxy behavior would change if the new no_proxy behavior would be enabled
+		// Test the new proxy function to see if the behavior will change in the future
 		newURL, _ := proxyConfig.ProxyFunc()(r.URL)
-		if url != newURL && url != nil {
-			urlString := r.URL.String()
-			NoProxyWarningMapMutex.Lock()
-			if _, ok := NoProxyWarningMap[urlString]; !ok {
-				NoProxyWarningMap[log.SanitizeURL(r.URL.String())] = true
-				logSafeURL := r.URL.Scheme + "://" + r.URL.Host
-				log.Warnf("Deprecation warning: the HTTP request to %s uses proxy %s but will ignore the proxy when the Agent configuration option no_proxy_nonexact_match defaults to true in a future agent version. Please adapt the Agent’s proxy configuration accordingly", logSafeURL, url.String())
-			}
-			NoProxyWarningMapMutex.Unlock()
+
+		if url == nil && newURL == nil {
+			return url, err
+		}
+
+		logSafeURL := r.URL.Scheme + "://" + r.URL.Host
+
+		// Print a warning if the url would ignore the proxy when no_proxy_nonexact_match is true
+		if url != nil && newURL == nil {
+			warnOnce(NoProxyIgnoredWarningMap, logSafeURL, "Deprecation warning: the HTTP request to %s uses proxy %s but will ignore the proxy when the Agent configuration option no_proxy_nonexact_match defaults to true in a future agent version. Please adapt the Agent’s proxy configuration accordingly", logSafeURL, url.String())
+			return url, err
+		}
+
+		var newURLString string
+		if newURL != nil {
+			newURLString = newURL.String()
```

Comment:
I don't think `newURLString` here is safe for logging (although the potential user/password credentials may be detected and scrubbed by default by our logger). Can you check?

Reply:
I think you are right. I changed it to only report the scheme + host just to be safe.